File: IMS/IMS.Presentation/DTOs/InternshipDTO.cs
using Shared.Enums;

namespace IMS.Presentation.DTOs;

public class InternshipDTO
{
    public required Guid InternId { get; set; } 

    public required Guid MentorId { get; set; } 

    public required Guid HumanResourcesManagerId { get; set; } 

    public DateTime StartDate { get; set; } = DateTime.UtcNow;

    public DateTime? EndDate { get; set; }

    public InternshipStatus Status { get; set; } = InternshipStatus.NotStarted;
}

File: Frontend/ims-frontend/src/entities/ims/dto/internship_dto.ts
import { InternshipStatus } from "../enums";

export interface InternshipDto {
    id: string;
    internId: string;
    mentorId: string;
    humanResourcesManagerId: string;
    startDate: string;
    endDate?: string | null;
    status: InternshipStatus;
}

export interface CreateInternshipDto {
    internId: string;
    mentorId: string;
    humanResourcesManagerId: string;
    startDate: string;
    endDate?: string | null;
    status: InternshipStatus;
}

export interface UpdateInternshipDto {
    mentorId: string;
    humanResourcesManagerId: string;
    startDate: string;
    endDate?: string | null;
    status: InternshipStatus;
}
